.github/workflows/diff.yml

```diff
@@ -14,67 +16,106 @@ jobs:
   build:
     name: Build
     needs: [get-runner-labels]
-    if: github.event.issue.pull_request && contains(github.event.comment.body, '!diff')
+    if: (github.event_name == 'issue_comment' || github.event_name == 'pull_request_review_comment') && contains(github.event.comment.body, '!diff')
     strategy:
-      fail-fast: false # Build and test everything so we can look at all the errors
       matrix:
         array:
           - target: x86_64-unknown-linux-gnu # For Cloud IDE
             runner: ${{ needs.get-runner-labels.outputs.LINUX_RUNNER_LABELS }}
     uses: ./.github/workflows/reusable-build.yml
     with:
       profile: "debug"
-      ref: refs/pull/${{ github.event.issue.number }}/head
+      ref: refs/pull/${{ github.event.issue.number || github.event.pull_request.number }}/head
       target: ${{ matrix.array.target }}
       runner: ${{ matrix.array.runner }}
       test: false
 
   diff:
     name: Diff Assets
-    runs-on: ubuntu-latest
-    needs: build
+    needs: [build, get-runner-labels]
+    runs-on: ${{ fromJSON(needs.get-runner-labels.outputs.LINUX_RUNNER_LABELS) }}
     steps:
       - name: Checkout Branch
         uses: actions/checkout@v3
         with:
           fetch-depth: 1
-          ref: refs/pull/${{ github.event.issue.number }}/head
+          ref: refs/pull/${{ github.event.issue.number || github.event.pull_request.number }}/head
 
       - name: Pnpm Cache
         uses: ./.github/actions/pnpm-cache
 
-      - name: Download artifacts
-        uses: actions/download-artifact@v3
+      - name: Download bindings
+        uses: ./.github/actions/download-artifact
         with:
-          path: artifacts
+          name: bindings-x86_64-unknown-linux-gnu
+          path: crates/node_binding/
+          try-local-cache: true
+          link-when-local: true
 
       - name: Build node packages
-        run: pnpm run build:js
-
-      - name: Move artifacts
-        run: node scripts/build-npm.cjs
-
-      - name: Build Viewer
-        run: pnpm run build:viewer
+        run: pnpm run build:js && pnpm run build:viewer
 
       - name: Run Diff
         run: node scripts/diff.cjs
 
       - name: Upload Report
-        uses: dmnemec/copy_file_to_another_repo_action@main
+        shell: bash
         env:
           API_TOKEN_GITHUB: ${{ secrets.RSPACK_REPORT_ACCESS_TOKEN }}
-        with:
-          source_file: diff_output/.
-          destination_repo: web-infra-dev/rspack-report-website
-          destination_folder: diff/${{ github.run_id }}/
-          user_email: lingyucoder@gmail.com
-          user_name: LingyuCoder
-          commit_message: Auto upload diff report
+        run: |
+          temp_dir=$(mktemp -d)
```

Comment:
mktemp always return a new empty directory, maybe we can set the cache_dir="$HOME/.cache/diff_upload"